feat: handle errors from `BundlerBuilder#build`

shulaoda opened the pull request:
In `BundlerBuilder#build`, some option-related errors or other issues cannot be represented as warnings, so we need to provide proper error handling for build failures.

shulaoda commented:
* **#6104** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6104?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6104?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6104).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hyf0 commented on crates/rolldown/src/bundler_builder.rs:
```diff
@@ -24,7 +24,7 @@ pub struct BundlerBuilder {
 
 impl BundlerBuilder {
-  pub fn build(mut self) -> Bundler {
+  pub fn build(mut self) -> BuildResult<Bundler> {
```
I didn't see any error returned?

shulaoda replied:
Yes, I plan to add the related errors in the next PR. This one only adds the functional support.

shulaoda replied:
done in this PR

shulaoda replied:
There are still some errors to handle, and we also need to support more user-friendly error messages.

shulaoda committed:
feat: handle errors from `BundlerBuilder#build` (#6104)

In `BundlerBuilder#build`, some option-related errors or other issues cannot be represented as warnings, so we need to provide proper error handling for build failures.
